rust/gui-client/src-tauri/src/client/updates.rs

```diff
@@ -101,168 +88,121 @@ pub(crate) async fn check() -> Result<Release, Error> {
         .await?;
     let status = response.status();
     if status != reqwest::StatusCode::OK {
-        return Err(Error::HttpStatus(status));
+        anyhow::bail!("HTTP status: {status}");
     }
 
     let response = response.text().await?;
-    Release::from_str(&response)
+    Ok(serde_json::from_str(&response)?)
```

Comment:
We could avoid hitting the GitHub API now, using the following algorithm:

1. Check what `https://www.firezone.dev/dl/firezone-client-windows-gui/latest/x86_64` redirects to
2. Parse version from redirected URL
3. If a new version is available, alert the user to download using this URL

Reply:
It may be a bit tricky to test. If you want to test locally you can try running the website locally:

```
cd website/
pnpm i
pnpm dev
```

Reply:
Yeah it got complicated. I'll make it a separate PR that stacks on this.